model/RolePrivilegeRetriever.php

```diff
@@ -58,6 +57,21 @@ public function retrieveByResourceIds(array $resourceIds): array
             $permissions[$user][] = $result['privilege'];
         }
 
+        return $this->removeDuplicatedEntries($permissions);
+    }
+
+    /**
+     * Remove possible duplicates that may come from merging ACLs from different
+     * resources after they've been merged together by user.
+     *
+     * @param string[][] $permissions
+     */
+    private function removeDuplicatedEntries(array $permissions)
+    {
+        foreach ($permissions as $_roleURI => &$entries) {
```

Comment:
Lets please not use `_` in front of variables as it is not common practice in TAO. Also, lets use camel case, please. I've never saw you writing code with this style. Maybe a side effect from python? :) 

```suggestion
        foreach ($permissions as $roleUri => &$entries) {
```

Reply:
Removed: 90aa60181a5df17e49864771843bcf60417c3c79